src/main/java/net/snowflake/ingest/streaming/internal/InternalStageManager.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import static net.snowflake.ingest.connection.ServiceResponseHandler.ApiName.STREAMING_CLIENT_CONFIGURE;
+import static net.snowflake.ingest.utils.Constants.BLOB_EXTENSION_TYPE;
+import static net.snowflake.ingest.utils.Constants.CLIENT_CONFIGURE_ENDPOINT;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.util.Calendar;
+import java.util.List;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import net.snowflake.client.jdbc.SnowflakeSQLException;
+import net.snowflake.ingest.utils.ErrorCode;
+import net.snowflake.ingest.utils.SFException;
+import net.snowflake.ingest.utils.Utils;
+
+/** Class to manage single Snowflake internal stage */
+class InternalStageManager<T> implements StorageManager<T> {
+  // Target stage for the client
+  private final StreamingIngestStorage targetStage;
+
+  // Increasing counter to generate a unique blob name per client
+  private final AtomicLong counter;
+
+  // Whether the manager in test mode
+  private final boolean isTestMode;
+
+  /**
+   * Constructor for InternalStageManager
+   *
+   * @param isTestMode whether the manager in test mode
+   * @param client the owning client
+   */
+  InternalStageManager(boolean isTestMode, SnowflakeStreamingIngestClientInternal<T> client) {
+    ConfigureCallHandler configureCallHandler =
+        ConfigureCallHandler.builder(
+                client.getHttpClient(),
+                client.getRequestBuilder(),
+                STREAMING_CLIENT_CONFIGURE,
+                CLIENT_CONFIGURE_ENDPOINT)
+            .setRole(client.getRole())
+            .setIsTestMode(isTestMode)
+            .build();
+    this.isTestMode = isTestMode;
+    this.counter = new AtomicLong(0);
+    try {
+      targetStage =
+          new StreamingIngestStorage(
+              isTestMode, configureCallHandler, client.getName(), DEFAULT_MAX_UPLOAD_RETRIES);
+    } catch (SnowflakeSQLException | IOException err) {
+      throw new SFException(err, ErrorCode.UNABLE_TO_CONNECT_TO_STAGE);
+    }
+  }
+
+  /**
+   * Get the storage. In this case, the storage is always the target stage as there's only one stage
+   * in non-iceberg mode.
+   *
+   * @param blobData this parameter does not affect the method outcome
+   * @return the target storage
+   */
+  @Override
+  @SuppressWarnings("unused")
+  public StreamingIngestStorage getStorage(List<List<ChannelData<T>>> blobData) {
+    // There's always only one stage for the client in non-iceberg mode
+    return targetStage;
+  }
+
+  /**
+   * Add storage to the manager. Should not be called in non-iceberg mode.
```

Comment:
comment appears untrue; it will still be called with the open channel API's response handling in IngestClientInternal.openChannel?

Reply:
Fixed.